nixos/modules/virtualisation/proxmox-lxc.nix

```diff
@@ -50,6 +59,8 @@ with lib;
         useDHCP = false;
         useHostResolvConf = false;
         useNetworkd = true;
+        # pick up hostname from /etc/hostname generated by proxmox
+        hostName = mkIf (!cfg.manageHostName) (mkForce "");
```

Comment:
```suggestion
        hostName = mkIf (!cfg.manageHostName) (default "");
```

Reply:
As mentioned in the first comment above,

> `mkDefault` will break with morph, and probably other deployment tools. Needs to be changed to an option just like `manageNetwork`.

This is because morph (and presumably nixops, though I've not tested this) sets the hostname to the machine name with mkDefault.

https://github.com/DBCDK/morph/blob/5b852370d8054a895b5ba79b5ef017c3afbb3a3c/data/eval-machines.nix#L28

Because this mkForce is wrapped in a mkIf, it is easy for anyone that wants the default behaviour to disable the mkForce.

Reply:
This is what happens if mkDefaultis used:
```
$ morph build targets/pvehostname/pvehostname.nix
Selected 1/1 hosts (name filter:-0, limits:-0):
          0: test (secrets: 0, health checks: 0, tags: )

error: The option `networking.hostName' has conflicting definition values:
       - In `/home/illustris/src/morph-test/profiles/pve_lxc.nix': ""
       - In `/nix/store/vbwfc1nyx078qymgv4mx18d6jqd0lvlr-morph-unstable-dev-lib/eval-machines.nix': "test"
(use '--show-trace' to show detailed location information)
Error while running `/run/current-system/sw/bin/nix-build /nix/store/vbwfc1nyx078qymgv4mx18d6jqd0lvlr-morph-unstable-dev-lib/eval-machines.nix --arg networkExpr /home/illustris/src/morph-test/targets/pvehostname/pvehostname.nix --argstr argsFile /run/user/1000/morph-3343142244/morph-args.json --out-link /run/user/1000/morph-3343142244/result --attr machines ...`: See above.
```

Reply:
Sounds to me more like we are fixing downstream bugs which should really be patched in the tools.

Reply:
`hostName = mkDefault machineName ` is a reasonable way for tools like nixops and morph to infer hostname. This isn't a fix for a downstream bug, but a sensible default for proxmox LXCs that accounts for proxmox's ability to set the hostname for containers. Do you have a better solution?

Reply:
ah, right. Now I understand.